pkg/registry/apis/dashboard/schema_validation.go

```diff
@@ -0,0 +1,71 @@
+package dashboard
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+
+	"github.com/grafana/grafana/apps/dashboard/pkg/apis/dashboard/v0alpha1"
+	"github.com/grafana/grafana/apps/dashboard/pkg/apis/dashboard/v1alpha1"
+	"github.com/grafana/grafana/apps/dashboard/pkg/apis/dashboard/v2alpha1"
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ValidateDashboardSpec validates the dashboard spec and throws a detailed error if there are validation errors.
+func ValidateDashboardSpec(ctx context.Context, obj runtime.Object, a admission.Attributes) error {
+	accessor, err := utils.MetaAccessor(obj)
+	if err != nil {
+		return fmt.Errorf("error getting meta accessor: %w", err)
+	}
+	mode := getFieldValidationMode(a)
+
+	if mode == metav1.FieldValidationIgnore {
+		// We don't want to validate the dashboard spec if the validation is set to ignore.
+		return nil
+	}
+
+	if mode == metav1.FieldValidationWarn {
+		// TODO: not sure how to return warnings
+		return nil
+	}
+
+	var errors field.ErrorList
+	switch v := obj.(type) {
+	case *v0alpha1.Dashboard:
+		// No-op for v0, we don't care about validating this API.
```

Comment:
Something that I didn't realize earlier – do we want to run the validation and log the error (maybe as INFO) never return it, just so that we can gauge how many dashboards we have in v0 with non-adhering schema?

Reply:
Added